server/csharp/TicketHub/Controllers/TicketController.cs

```diff
@@ -33,23 +40,89 @@ public TicketController(ILogger<TicketController> logger, PostgresContext dbCont
+            foreach (var fieldInfo in fieldInfos)
+            {
+                fieldMapping[fieldInfo.Name] = fieldInfo;
+            }
+
+            mapping[entityType.Name] = fieldMapping;
+        }
+
+        return mapping;
+    }
+
     // List all tickets.
     [HttpGet]
     [Route("tickets")]
     [OpaRuleAuthorization("tickets/allow", "list")]
     public async Task<ActionResult<IAsyncEnumerable<Ticket>>> ListTickets()
     {
+        //_logger.LogInformation("{0}", JsonConvert.SerializeObject(result));
         var tName = HttpContext.Items["Tenant"]?.ToString();
+        string subject = HttpContext.Items["Subject"]?.ToString() ?? "";
+        //var mapper = TicketExpressionMapper.CreateMapping();
+        var userExpr = Expression.Parameter(typeof(User), "user");
+        var ticketExpr = Expression.Parameter(typeof(Ticket), "ticket");
+
+        // TODO: Refactor the QueryableExtension stuff to allow mapping from a source parameter expression?
+        // Ex: Something like Func<ParameterExpression, string> -> PropertyExpression
+        // These mappings will need to exist for each possible model that queries would be based off of.
+        // It's inconvenient, but that's the price we pay for building on top of LINQ, as best I can tell.
+
+        var mapper = new Dictionary<string, Func<ParameterExpression, Expression>>()
+        {
+            { "tickets.id", t => Expression.Property(t, "Id") },
+            { "tickets.customer", t => Expression.Property(t, "Customer") },
+            { "tickets.customer.name", t => Expression.Property(t, "CustomerName") },
+            { "tickets.tenant", t => Expression.Property(t, "Tenant") },
+            { "tickets.tenant.name", t => Expression.Property(t, "TenantName") },
+            { "tickets.assignee", t => Expression.Property(t, "Assignee") },
+            { "tickets.assignee.name", t => Expression.Property(t, "UserName") },
+            { "tickets.description", t => Expression.Property(t, "Description") },
+            { "tickets.resolved", t => Expression.Property(t, "Resolved") },
+            { "tickets.last_updated", t => Expression.Property(t, "LastUpdated") },
+            { "users.id", t => Expression.Property(Expression.Property(t, "UserNavigation"), "Id") },
+            { "users.name", t => Expression.Property(Expression.Property(t, "UserNavigation"), "Name") },
+            { "users.tenant", t => Expression.Property(Expression.Property(t, "UserNavigation"), "Tenant") },
+        };
         if (tName is string)
         {
             Tenant tenant = await getTenantByName(tName);
+            var conditions = await getConditions(HttpContext, "tickets_expanded/response", new Dictionary<string, object>(){
+                { "tenant", tenant },
+                { "user", subject },
+                { "action", "list" },
+            });
+            if (conditions is null)
+            {
+                return StatusCode(500, "No tickets found");

```

Comment:
I'd have expected `StatusCode(404, "No tickets found");` -- am I mislead in some way? 🤔

Reply:
Hmmmmm... That looks suspicious. I'll refactor that.